src/components/Footer/Footer.js

```diff
@@ -28,8 +39,21 @@ const Footer = ({
   blockStatus,
   loadingVote,
   showScoreButton = false,
-  onPressScore
+  onPressScore,
+  isShowDM = false
 }) => {
+  const {sendMessageDM} = useDMMessage();
+  const [profile] = React.useContext(Context).profile;
+  const [userAllowDm, setUserAllowDm] = React.useState(false);
+
+  const [loading, setLoading] = React.useState({
+    loadingDm: false,
+    loadingDmAnon: false,
+    loadingGetAllowAnonDmStatus: false
+  });
+  const refSheet = React.useRef();
+  const {createSignChat} = useCreateChat();
+
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

New state variables and hooks are introduced here. The `useState` hook is used to manage local states for loading statuses and user permissions. The `useRef` hook is used to create a reference to the BottomSheetMenu component. The `useDMMessage` and `useCreateChat` hooks are used to handle direct messages and chat creation respectively. These changes seem appropriate and follow React best practices.


<!-- This is an auto-generated comment by OSS CodeRabbit -->